test/ServiceRegistryL2.js

```diff
@@ -0,0 +1,1167 @@
+            ).to.be.revertedWith("ZeroValue");
+
+            await serviceRegistry.setBaseURI("https://localhost2/service/");
+            expect(await serviceRegistry.baseURI()).to.equal("https://localhost2/service/");
+        });
+    });
+
+    context("Service creation", async function () {
+        it("Should fail when creating a service without a serviceManager", async function () {
+            const owner = signers[3].address;
+            await expect(
+                serviceRegistry.create(owner, configHash, agentIds, agentParams, threshold)
+            ).to.be.revertedWith("ManagerOnly");
+        });
+
+        it("Should fail when changing the owner and the manager to a zero address", async function () {
+            // Try to change the owner to the zero address
+            await expect(
+                serviceRegistry.changeOwner(AddressZero)
+            ).to.be.revertedWith("ZeroAddress");
+
+            // Try to change the manager to the zero address
+            await expect(
+                serviceRegistry.changeManager(AddressZero)
+            ).to.be.revertedWith("ZeroAddress");
+        });
+
+        it("Should fail when the owner of a service has a zero address", async function () {
+            const serviceManager = signers[3];
+            await serviceRegistry.changeManager(serviceManager.address);
+            await expect(
+                serviceRegistry.connect(serviceManager).create(AddressZero, configHash, agentIds,
+                    agentParams, threshold)
+            ).to.be.revertedWith("ZeroAddress");
+        });
+
+        it("Should fail when creating a service with a zero value config hash", async function () {
+            const serviceManager = signers[3];
+            const owner = signers[4].address;
+            await serviceRegistry.changeManager(serviceManager.address);
+            await expect(
+                serviceRegistry.connect(serviceManager).create(owner, ZeroBytes32, agentIds, agentParams,
+                    threshold)
+            ).to.be.revertedWith("ZeroValue");
+        });
+
+        it("Should fail when creating a service with incorrect agent slots values", async function () {
+            const serviceManager = signers[3];
+            const owner = signers[4].address;
+            await serviceRegistry.changeManager(serviceManager.address);
+            await expect(
+                serviceRegistry.connect(serviceManager).create(owner, configHash, [], [], threshold)
+            ).to.be.revertedWith("WrongArrayLength");
+            await expect(
+                serviceRegistry.connect(serviceManager).create(owner, configHash, [1], [], threshold)
+            ).to.be.revertedWith("WrongArrayLength");
+            await expect(
+                serviceRegistry.connect(serviceManager).create(owner, configHash, [1, 3], [[2, regBond]],
+                    threshold)
+            ).to.be.revertedWith("WrongArrayLength");
+        });
+
```

Comment:
what do we expect to happen when a service is created with a non-existent canonical agent?

Reply:
Ok, so not sure if it makes sense to add somewhere that whether the service is stuck to the agent instance registration, it is possible that one of the canonical agents is non-existent and if the latter is verified the service can be updated with the right ids.

Reply:
We have stated that at the top of the contract